packages/core/admin/admin/src/utils/getFetchClient.ts

```diff
@@ -1,67 +1,53 @@
-import axios, { AxiosRequestConfig, AxiosResponse, AxiosInstance } from 'axios';
 import qs from 'qs';
 
 const STORAGE_KEYS = {
   TOKEN: 'jwtToken',
   USER: 'userInfo',
 };
 
-const fetchClient = (): AxiosInstance => {
-  const instance = axios.create({
-    headers: {
-      Accept: 'application/json',
-      'Content-Type': 'application/json',
-    },
-    paramsSerializer: (params) => {
-      return qs.stringify(params, { encode: false });
-    },
-  });
+type FetchParams = Parameters<typeof fetch>;
+type FetchURL = FetchParams[0];
+export type FetchOptions = FetchParams[1];
 
-  // Add a request interceptor to add authorization token to headers, rejects errors
-  instance.interceptors.request.use(
-    async (config) => {
-      config.headers.Authorization = `Bearer ${getToken()}`;
-
-      return config;
-    },
-    (error) => Promise.reject(error)
-  );
+export type FetchResponse<TData = unknown> = {
+  data: TData;
+};
+type Options = {
+  params?: any;
+  baseURL?: string;
+};
+export type FetchConfig = {
+  options?: Options;
+  fetchConfig?: FetchParams[1];
+};
```

Comment:
it could, I just think it would be confusing, the fetchConfig is basically direct access to the native fetch options, something don't have control over essentially we don't know what new options could be added in the future, and possibility of name clashes between things we implemented ourselves and things that are provided.
One other thing I thought about is that we could hide access to the native fetch options completely and provide our own custom interface for options that we find now needed, and that would also help hide the implementation details away

Reply:
Yeah if you want to hide the implementation detail i would add config options for the things we allow but it keep it flatter to it's a simpler API surface 👍🏼